File: examples/python/bad/lsp-2.py
class Bird:
    def eat(self, food):
        print(f"I can eat {food}")

    def fly(self):
        print("Fly, Fly!!")

class Duck(Bird):
    def fly(self):
        print("I quack, quack while flying!!")

class Chicken(Bird):
    def fly(self):
        raise Exception("I cannot fly!!")


birds = [Duck(), Chicken()]
for bird in birds:
    bird.fly()

File: examples/typescript/bad/lsp-2.js
"use strict";
class Bird {
    eat(food) {
        console.log(`I can eat ${food}`);
    }
    fly() {
        console.log("Fly!! Fly!!");
    }
}
class Duck extends Bird {
    fly() {
        console.log("I quack, quack while flying!!");
    }
}
class Chicken extends Bird {
    fly() {
        console.error("I cannot fly!!");
    }
}
(function () {
    const birds = [new Duck(), new Chicken()];
    for (const bird of birds) {
        bird.fly();
    }
})();
